src/Serializer/ItemNormalizer.php

```diff
@@ -24,43 +22,4 @@
  */
 class ItemNormalizer extends AbstractItemNormalizer
 {
-    /**
-     * {@inheritdoc}
-     *
-     * @throws InvalidArgumentException
-     */
-    public function denormalize($data, $class, $format = null, array $context = [])
-    {
-        // Avoid issues with proxies if we populated the object
-        if (isset($data['id']) && !isset($context[self::OBJECT_TO_POPULATE])) {
-            if (isset($context['api_allow_update']) && true !== $context['api_allow_update']) {
-                throw new InvalidArgumentException('Update is not allowed for this operation.');
-            }
-
-            $this->updateObjectToPopulate($data, $context);
-        }
-
-        return parent::denormalize($data, $class, $format, $context);
-    }
-
-    private function updateObjectToPopulate(array $data, array &$context)
-    {
-        try {
-            $context[self::OBJECT_TO_POPULATE] = $this->iriConverter->getItemFromIri((string) $data['id'], $context + ['fetch_data' => true]);
-        } catch (InvalidArgumentException $e) {
```

Comment:
this has to be retained, it fixes https://github.com/api-platform/core/issues/857